tensorflow/python/ops/gradients_util.py

```diff
@@ -542,9 +542,6 @@ def _GradientsHelper(ys,
                      unconnected_gradients=UnconnectedGradients.NONE,
                      src_graph=None):
   """Implementation of gradients()."""
-  if context.executing_eagerly():
```

Comment:
I think this still applies. This PR makes V2 optimizers step into graph mode to get updates (and `context.executing_eagerly()` will return False when in that scope), but the restriction is still present in general.

Reply:
Yeah I agree - reinstated the check.